pkg/availability/multisigcollector/internal/parts/certverification/certverification.go

```diff
@@ -148,7 +154,8 @@ func verifyCertificateStructure(params *common.ModuleParams, cert *apbtypes.Cert
 		}
 
 		// Check that signers are members.
-		if !sliceutil.ContainsAll(params.AllNodes, mscCert.Signers) {
+		// TODO: This check can be extremely inefficient (quadratic in membership size), especially at large scale.
+		if !sliceutil.ContainsAll(maputil.GetKeys(params.Membership.Nodes), mscCert.Signers) {
```

Comment:
Good point, gladly we can just change the code of the containsAll function. We can make it linear using a completely reasonable extra amount of space with a map:
```
// ContainsAll returns true if the slice contains all elements of the given slice.
func ContainsAll[T comparable](ts []T, ts2 []T) bool {
	elements := make(map[T]struct{})
	for _, t := range ts {
		elements[t] = struct{}{}
	}

	for _, t := range ts2 {
		if _, ok := elements[t]; !ok {
			return false
		}
	}
	return true
}
```

(Can be made in this PR if you want or later on, but then it would be good to add an issue to the backlog at least)

Reply:
Yes, let's make an issue of that.